common/src/enforcer/source/cargo/mod.rs

```diff
@@ -0,0 +1,55 @@
+use crate::enforcer::{source::Source, Dependency};
+use crate::highlight::Range;
+use anyhow::anyhow;
+use async_trait::async_trait;
+use std::path::PathBuf;
+use url::Url;
+
+fn cargo_to_purl(name: String, dep: cargo_toml::Dependency) -> Option<Dependency> {
+    match dep {
+        cargo_toml::Dependency::Simple(version) => Some(Dependency {
+            purl: Url::parse(format!("pkg:cargo/{name}@{version}").as_str()).unwrap(),
```

Comment:
Are is some more information which we should add (such as git dependencies, patches, custom registry). This information too is available from the lock file.